Created an audio player that plays my favourite music.

eriyezab opened the pull request:
The music files are in mp3 format and are located in a directory named audio. Apologies if spelling favourite like this bothers you as I am Canadian lol

eriyezab committed:
Add audio directory with 5 of favourite songs in mp3 format.

eriyezab committed:
Add audio player and list of links to favourite songs in music section.

eriyezab committed:
Add audio player that plays songs from audio directory when song in list is clicked.

eriyezab committed:
Add autoplay when song is clicked.

eriyezab committed:
Merge branch 'FactFeature' of https://github.com/eriyezab/my-portfolio into AudioPlayer

eriyezab committed:
Merge branch 'master' of https://github.com/eriyezab/my-portfolio into AudioPlayer

shrutip90 commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,26 @@ function addRandomFact() {
+  currentSong = 0;
+  player = document.getElementById('audio-player');
+  songs = document.querySelectorAll('#song-list li a');
+  console.log('Adding Event Listener to song links...');
```
This doesn't seem very useful unless you expect the addEventListener code to fail or not run for some reason.

eriyezab replied:
When I was making the function I was running into issues and so I had added it there in order to debug and forgot to take it out. I realized that the problem was that inside the index.html the script tag was at the top and so it was not working since the document hadn't loaded.

shrutip90 commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,26 @@ function addRandomFact() {
+    song.addEventListener('click', (event) => {
+      console.log('Clicked song.');
+      event.preventDefault();
+      songList[currentSong].parentNode.className = '';
```
Can you use index here instead?

shrutip90 replied:
Nvm, I guess that wouldn't work.

shrutip90 commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,26 @@ function addRandomFact() {
+    song.addEventListener('click', (event) => {
+      console.log('Clicked song.');
+      event.preventDefault();
+      songList[currentSong].parentNode.className = '';
```
Just curious, why is this required?

eriyezab replied:
This was so that when I change the song, the previous song no longer had a class name of current song and I was planning on styling it such that the current song was in a different colour than the others but I hadn't gotten to that yet. I should have made more comments clarifying this

shrutip90 replied:
Okay, thanks. One suggestion would be to avoid adding changes that are unused currently - like you could add the 'current-song' class in the pull request where you are also adding the styling, so that it makes more sense :).

eriyezab committed:
Add clarifying comments to the func audioPlayer

Slightly re arranged the code and added clarifying comments to describe the behaviour of the function and why certain lines were necessary.

shrutip90 reviewed (approved):
Thanks for adding more comments!
